### PR TITLE
Correct parentheses for [`needless_borrow`] suggestion

### LINKED ISSUE #12268: Unnecessary parentheses in suggested fix for `needless_borrow`
### Summary

After updating to the latest nightly, `cargo clippy --fix` makes an annoying-looking fix with extraneous parentheses.

### Reproducer

I ran `cargo clippy --workspace` in https://github.com/martinvonz/jj/tree/afebea6e730e853ecefb856d1c850612f146c93c. (Sorry, I know this is not minimal).

I got the following warnings. The warnings are correct, but the suggested fix isn't.

```
warning: this expression creates a reference which is immediately dereferenced by the compiler
   --> lib/src/repo_path.rs:138:33
    |
138 | ..._or_else(|| (&self.value, &self.value[self.v...
    |                 ^^^^^^^^^^^ help: change this to: `(self.value)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
    = note: `#[warn(clippy::needless_borrow)]` on by default

warning: this expression creates a reference which is immediately dereferenced by the compiler
   --> lib/src/repo_path.rs:152:51
    |
152 | ....value[..0], &self.value));
    |                 ^^^^^^^^^^^ help: change this to: `(self.value)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

warning: `jj-lib` (lib) generated 2 warnings (run `cargo clippy --fix --lib -p jj-lib` to apply 2 suggestions)
```

#### I expected to see this happen:

The fix in both cases should be `self.value`, not `(self.value)`.

#### Instead, this happened:

See above. Running `cargo clippy --fix` also applies the weird suggestion, which is then annoying to clean up.

### Version

```text
rustc 1.78.0-nightly (d44e3b95c 2024-02-09)
binary: rustc
commit-hash: d44e3b95cb9d410d89cb8ab3233906a33f43756a
commit-date: 2024-02-09
host: aarch64-apple-darwin
release: 1.78.0-nightly
LLVM version: 17.0.6



clippy 0.1.78 (d44e3b95 2024-02-09)
```

### Additional Labels

_No response_

### PR DESCRIPTION
**Pull request by mira-eanda**
This fixes #12268 

Clippy no longer adds unnecessary parentheses in suggestions when the expression is part of a tuple.

---

changelog: Fix [`needless_borrow`] unnecessary parentheses in suggestion.

### CODE REVIEW

File: clippy_lints/src/dereference.rs [hunk 1]
```diff
@@ -1014,9 +1014,15 @@ fn report<'tcx>(
                         },
                         _ => (0, false),
                     };
+                    let is_in_tuple = match cx.tcx.parent_hir_node(data.first_expr.hir_id) {
+                        Node::Expr(e) => matches!(e.kind, ExprKind::Tup(_)),
```
Comment (J-ZhengLi) [inline clippy_lints/src/dereference.rs:1018]: nit: use `clippy_utils::get_parent_expr(cx, data.first_expr)`

Which does exactly the same thing.

File: tests/ui/needless_borrow.rs [hunk 2]
```diff
@@ -251,3 +251,10 @@ mod issue_10253 {
         (&S).f::<()>();
     }
 }
+
+fn issue_12268() {
+    let option = Some((&1,));
+    let x = (&1,);
+    // Lint here.
+    option.unwrap_or((&x.0,));
```
Comment (m-rph) [inline tests/ui/needless_borrow.rs:259]: It's generally better to use tagging to indicate that lints should be present. This helps prevent regressions.
```suggestion

    option.unwrap_or((&x.0,));
    //~^ ERROR: this expression creates a reference which
```

For more details on tagging:
https://doc.rust-lang.org/clippy/development/writing_tests.html#write-test-cases

**Review by m-rph — COMMENTED** [review]
Just a nit on the tests.

File: clippy_lints/src/dereference.rs (hunk 1 above)
Comment (Manishearth) [reply clippy_lints/src/dereference.rs:1018]: (waiting for this)

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
@bors r+

thanks!

**Pull request merged**